[Java] Add visibility check to selectByVisibleText in Select class

SantiagoCecena opened the pull request:
### **User description**
### 🔗 Related Issues
Fixes #15265

### 💥 What does this PR do?
This PR updates the `selectByVisibleText` method in the Java `Select` class to prevent selecting `<option>` elements that are hidden via CSS.

Specifically, the method now checks for CSS properties:
- `display: none`
- `visibility: hidden`
- `opacity: 0` or `0.0`

An existing private method, `hasCssPropertyAndVisible`, is used to verify visibility.

### 🔧 Implementation Notes
The fix was done by calling `assertSelectIsVisible()` at the beginning of the method, following the same pattern already used in `selectByContainsVisibleText`.

No new logic or dependencies were introduced, and behavior is now consistent across both selection methods.

### 💡 Additional Considerations
The change was manually tested in a local environment using WebDriver, since I currently cannot run Bazel locally due to company machine restrictions.

If needed, I'm happy to follow up with a Bazel-compatible test or collaborate with a maintainer to integrate one.

### 🔄 Types of changes
- Bug fix (backwards compatible)


___

### **PR Type**
Bug fix


___

### **Description**
• Add visibility check to `selectByVisibleText` method in Java Select class
• Prevent selection of hidden options with CSS properties like `display: none`
• Make behavior consistent with `selectByContainsVisibleText` method


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Select.java</strong><dd><code>Add visibility validation to selectByVisibleText method</code>&nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/support/ui/Select.java

• Added <code>assertSelectIsVisible()</code> call at method start<br> • Added <br>visibility check for each option using <code>hasCssPropertyAndVisible()</code><br> • <br>Throw <code>NoSuchElementException</code> for invisible options with descriptive <br>message


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/15912/files#diff-255eaa03e269acb38bc85bd146a129dd90e1ae3e20d94fc6de98def15cc4e59f">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

SantiagoCecena committed:
Add visibility check to selectByVisibleText in Select class

This change updates the selectByVisibleText method to prevent
selecting <option> elements that are hidden via CSS properties:
"display: none", "visibility: hidden", and "opacity: 0" or "0.0".

It uses the existing hasCssPropertyAndVisible method to ensure
that only visible options can be selected, making this method
consistent with selectByContainsVisibleText, which already has
this behavior.

Fixes #15265

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/selenium?pullRequest=15912) <br/>All committers have signed the CLA.

cgoldberg commented:
@SantiagoCecena please fix the formatting and update your branch (missing space after `if`)

cgoldberg commented:
Also, please sign the CLA: https://github.com/SeleniumHQ/selenium/pull/15912#issuecomment-2986935121

SantiagoCecena committed:
style: format code to follow style guidelines

SantiagoCecena committed:
fix: continue searching for visible optiones before throwing in selectByVisibleText

diemol reviewed (changes requested):
@SantiagoCecena, there are test failures related to this change. Could you please check them?

SantiagoCecena commented:
@diemol but the test failures are in Ruby, right?  I worked in a feature in Java, so I don't understand why is this happening 🤔

diemol commented:
The ruby failures are isolated. 

The Java failures are here: https://github.com/SeleniumHQ/selenium/actions/runs/15824432595/job/44601350705?pr=15912

SantiagoCecena committed:
feat(java): handle invisible options gracefully in selectByVisibleText

Refactored the logic in selectByVisibleText to skip invisible options
instead of failing fast. The method now continues evaluating other
options with matching visible text, ensuring consistent behavior when
multiple matching elements exist but some are hidden via CSS.

This change improves robustness and aligns with cross-language
expectations for Select component behavior.

Relates to #15265

SantiagoCecena committed:
refactor(java): remove comment line in selectByVisibleText

diemol committed:
Merge branch 'trunk' into fix-java-select-visibility

diemol commented:
This PR (#16220) was merged. I am closing this one.
